File: springboot-backend/src/main/java/net/saravana/services/VendorService.java
package net.saravana.services;

import net.saravana.entities.Vendor;
import net.saravana.exceptions.ResourceNotFoundException;
import net.saravana.repositories.VendorRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class VendorService {
    @Autowired
    private final VendorRepository vendorRepository;

    public VendorService(VendorRepository vendorRepository) {
        this.vendorRepository = vendorRepository;
    }

    public List<Vendor> getAllVendors(){
        return vendorRepository.findAll();
    }
    public Vendor createVendor(Vendor vendor){
        return  vendorRepository.save(vendor);
    }

    public Vendor getVendorById(Long id){
        return vendorRepository.findById(id)
                .orElseThrow(() -> new ResourceNotFoundException("Vendor not exist with id :" + id));
    }
    public void deleteVendor(Vendor vendor){
        vendorRepository.delete(vendor);
    }
}

File: react-frontend/src/services/VendorService.js
import {request} from '../axios_helper';

const VENDOR_API_URL = "/api/v2/vendors";

class VendorService {

    getVendors(){
        return request("GET", VENDOR_API_URL, {});
    }

    createVendor(vendor){
        return request("POST", VENDOR_API_URL, vendor);
    }

    getVendorById(vendorId){
        return request("GET", VENDOR_API_URL + '/' + vendorId, {});
    }

    updateVendor(vendor, vendorId){
        return request("PUT", VENDOR_API_URL + '/' + vendorId, vendor);
    }

    deleteVendor(vendorId){
        return request("DELETE", VENDOR_API_URL + '/' + vendorId, {});
    }
}

export default new VendorService()